### PR TITLE
[Feature] Support `cast` to an array.

### PR DESCRIPTION
**Pull request by d0cd**
This PR allows users to initialize arrays with the `cast` instruction.
e.g. `cast r0 r1 r2 into r3 as [boolean; 3u32]`, `cast r3 into r4 as [[boolean; 3u32]; 1u32]`.

### CODE REVIEW

**Comment by raychu86** [comment]
Is there any case where interleaving structs and arrays and continuously nesting them, can get past the `N::MAX_DATA_DEPTH` checks?

File: synthesizer/program/src/logic/instruction/operation/cast.rs [hunk 1]
```diff
@@ -46,6 +46,7 @@ use console::{
     types::Field,
 };
 
+use console::program::ArrayType;
```
Comment (raychu86) [inline synthesizer/program/src/logic/instruction/operation/cast.rs]: Nit: Move this import up

File: synthesizer/program/src/logic/instruction/operation/cast.rs [hunk 2]
```diff
@@ -392,8 +393,44 @@ impl<N: Network> Cast<N> {
...
                 registers.store_circuit(stack, &self.destination, circuit::Value::Plaintext(struct_))
             }
-            CastType::RegisterType(RegisterType::Plaintext(PlaintextType::Array(..))) => {
-                bail!("Illegal operation: Cannot cast to an array.")
+            CastType::RegisterType(RegisterType::Plaintext(PlaintextType::Array(array_type))) => {
+                // Ensure the operands length is at least the minimum.
+                if inputs.len() < N::MIN_ARRAY_ELEMENTS {
+                    bail!("Casting to an array requires at least {} operand", N::MIN_ARRAY_ELEMENTS)
+                }
+
```
Comment (raychu86) [inline synthesizer/program/src/logic/instruction/operation/cast.rs:409]: Add check for `MAX_ARRAY_ELEMENTS`.

```suggestion
// Ensure the number of elements does not exceed the maximum.
if inputs.len() > N::MIN_ARRAY_ELEMENTS {
    bail!("'{array_type}' cannot exceed {} elements", N::MAX_ARRAY_ELEMENTS)
}

```

File: synthesizer/program/src/logic/instruction/operation/cast.rs [hunk 3]
```diff
@@ -392,8 +393,44 @@ impl<N: Network> Cast<N> {
                 // Store the struct.
                 registers.store_circuit(stack, &self.destination, circuit::Value::Plaintext(struct_))
             }
-            CastType::RegisterType(RegisterType::Plaintext(PlaintextType::Array(..))) => {
-                bail!("Illegal operation: Cannot cast to an array.")
+            CastType::RegisterType(RegisterType::Plaintext(PlaintextType::Array(array_type))) => {
+                // Ensure the operands length is at least the minimum.
+                if inputs.len() < N::MIN_ARRAY_ELEMENTS {
+                    bail!("Casting to an array requires at least {} operand", N::MIN_ARRAY_ELEMENTS)
```
Comment (raychu86) [inline synthesizer/program/src/logic/instruction/operation/cast.rs]: nit

```suggestion
                    bail!("Casting to an array requires at least {} operand(s)", N::MIN_ARRAY_ELEMENTS)
```

File: synthesizer/process/src/stack/finalize_types/matches.rs [hunk 4]
```diff
@@ -13,6 +13,7 @@
 // limitations under the License.
 
 use super::*;
+use console::program::ArrayType;
```
Comment (raychu86) [inline synthesizer/process/src/stack/finalize_types/matches.rs]: Nit: Move this import into the `super::*`

**Review by howardwu — APPROVED** [review]
LGTM

**Pull request merged**